src/lib/operators/aggregate_sort.cpp

```diff
@@ -583,16 +590,24 @@ std::shared_ptr<const Table> AggregateSort::_on_execute() {
          */
         const auto& value = (*segment)[group_start.chunk_offset];
 
-        null_values[value_index] = variant_is_null(value);
-        if (!null_values[value_index]) {
-          // Only store non-null values
+        if (column_is_nullable) {
+          null_values[value_index] = variant_is_null(value);
+          if (!null_values[value_index]) {
+            // Only store non-null values
+            values[value_index] = boost::get<ColumnDataType>(value);
+          }
+        } else {
           values[value_index] = boost::get<ColumnDataType>(value);
         }
       }
```

Comment:
Why make it this complicated? The vector has been initialized to false anyway.

```
if (column_is_nullable && variant_is_null(value)) {
  null_values[...] = true;
} else {
  values[...] = ...;
}
```

Am I missing a case?

Reply:
we can even drop the column_is_nullable check, since this PR makes sure that variant_is_null can only be true if column_is_nullable ;)